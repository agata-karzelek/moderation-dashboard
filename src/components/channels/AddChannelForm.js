import React from "react";
import { useStyles } from "../../style/createModerationModal";
import {
  DialogContent,
  DialogActions,
  Button,
  Typography,
  TextField,
  CircularProgress,
  Box,
} from "@material-ui/core";

export default function AddChannelForm(props) {
  const classes = useStyles();
  return (
    <form onSubmit={props.formik.handleSubmit}>
      <DialogContent>
        <Typography testid="Channel_Id" className={classes.subHeadingFont}>
          Channel ID
        </Typography>
        <TextField
          name="id"
          id="id"
          disabled={props.disabled}
          placeholder="Enter Channel ID"
          variant="outlined"
          className={classes.inputBox}
          autoComplete="off"
          onChange={props.formik.handleChange}
          value={props.formik.values.id}
          error={props.formik.touched.id && Boolean(props.formik.errors.id)}
          helperText={props.formik.touched.id && props.formik.errors.id}
        />
        <Box pt={1}>
          <Typography testid="Display_Name" className={classes.subHeadingFont}>
            Display Name
          </Typography>
          <TextField
            id="name"
            name="name"
            placeholder="Enter Display Name"
            variant="outlined"
            className={classes.inputBox}
            autoComplete="off"
            onChange={props.formik.handleChange}
            value={props.formik.values.name}
            error={props.formik.touched.name && Boolean(props.formik.errors.name)}
            helperText={props.formik.touched.name && props.formik.errors.name}
          />
        </Box>
        <Box pt={1}>
          <Typography testid="Description" className={classes.subHeadingFont}>
            Description
          </Typography>
          <TextField
            id="description"
            name="description"
            placeholder="Enter Channel Description"
            variant="outlined"
            className={classes.inputBox}
            multiline
            rows={3}
            autoComplete="off"
            onChange={props.formik.handleChange}
            value={props.formik.values.description}
            error={props.formik.touched.description && Boolean(props.formik.errors.description)}
            helperText={props.formik.touched.description && props.formik.errors.description}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button autoFocus className={classes.cancelButton} onClick={props.handleClose}>
          CANCEL
        </Button>
        <Button
          autoFocus
          className={classes.addButton}
          type="submit"
          color="primary"
          disabled={props.channelAlert.loading}
          startIcon={
            props.channelAlert.loading ? (
              <CircularProgress className={classes.loader} size={30} />
            ) : null
          }
        >
          <Typography className={classes.buttonText}>{props.buttonTitle}</Typography>
        </Button>
      </DialogActions>
    </form>
  );
}
