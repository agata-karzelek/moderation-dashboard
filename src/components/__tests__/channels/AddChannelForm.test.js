import React from "react";
import { shallow } from "enzyme";
import AddChannelForm from "../../channels/AddChannelForm";

const props = {
  formik: {
    handleSubmit: jest.fn(),
    disabled: false,
    values: {
      id: 3,
    },
    touched: {
      id: false,
    },
  },
  channelAlert: {
    loading: false,
  },
};

describe("Add User Test Cases", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<AddChannelForm formik={props.formik} channelAlert={props.channelAlert} />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Labels and initial Values", () => {
    expect(wrapper.find("[testid='Channel_Id']").text()).toBe("Channel ID");
    expect(wrapper.find("[testid='Display_Name']").text()).toBe("Display Name");
    expect(wrapper.find("[testid='Description']").text()).toBe("Description");
    expect(wrapper.find("[id='id']").text()).toBe("");
    expect(wrapper.find("[id='name']").text()).toBe("");
    expect(wrapper.find("[id='description']").text()).toBe("");
  });

  test("check placeholder", () => {
    expect(wrapper.find("[id='id']").getElement().props.placeholder).toBe("Enter Channel ID");
    expect(wrapper.find("[id='name']").getElement().props.placeholder).toBe("Enter Display Name");
    expect(wrapper.find("[id='description']").getElement().props.placeholder).toBe(
      "Enter Channel Description"
    );
  });
});
