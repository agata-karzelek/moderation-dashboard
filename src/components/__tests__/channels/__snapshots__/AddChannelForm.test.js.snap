// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add User Test Cases Snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-subHeadingFont-12"
      testid="Channel_Id"
    >
      Channel ID
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      autoComplete="off"
      className="makeStyles-inputBox-6"
      error={false}
      helperText={false}
      id="id"
      name="id"
      placeholder="Enter Channel ID"
      value={3}
      variant="outlined"
    />
    <Styled(MuiBox)
      pt={1}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-subHeadingFont-12"
        testid="Display_Name"
      >
        Display Name
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(TextField))
        autoComplete="off"
        className="makeStyles-inputBox-6"
        id="name"
        name="name"
        placeholder="Enter Display Name"
        variant="outlined"
      />
    </Styled(MuiBox)>
    <Styled(MuiBox)
      pt={1}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-subHeadingFont-12"
        testid="Description"
      >
        Description
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(TextField))
        autoComplete="off"
        className="makeStyles-inputBox-6"
        id="description"
        multiline={true}
        name="description"
        placeholder="Enter Channel Description"
        rows={3}
        variant="outlined"
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      className="makeStyles-cancelButton-9"
    >
      CANCEL
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      className="makeStyles-addButton-14"
      color="primary"
      disabled={false}
      startIcon={null}
      type="submit"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-buttonText-5"
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</form>
`;
